courses/ocp4_advanced_deployment/lab_08_1/solve_lab.yml

```diff
@@ -0,0 +1,234 @@
+      k8s:
+        state: present
+        definition:
+          apiVersion: operators.coreos.com/v1alpha1
+          kind: Subscription
+          metadata:
+            name: prometheus
+            namespace: my-prometheus
+          spec:
+            channel: beta
+            name: prometheus
+            source: community-operators
+            sourceNamespace: openshift-marketplace
+
+    - name: Create Prometheus Service Monitor
+      k8s:
+        state: present
+        definition:
+          apiVersion: monitoring.coreos.com/v1
+          kind: ServiceMonitor
+          metadata:
+            name: pod-autoscale
+            namespace: my-prometheus
+            labels:
+              lab: custom-hpa
+          spec:
+            namespaceSelector:
+              matchNames:
+                - my-prometheus
+                - my-hpa
+            selector:
+              matchLabels:
+                app: pod-autoscale
+            endpoints:
+            - port: 8080-tcp
+              interval: 30s
+
+    - name: Create Prometheus instance
+      k8s: 
+        state: present
+        definition:
+          apiVersion: monitoring.coreos.com/v1
+          kind: Prometheus
+          metadata:
+            name: my-prometheus
+            labels:
+              prometheus: my-prometheus
+            namespace: my-prometheus
+          spec:
+            replicas: 2
+            serviceAccountName: prometheus-k8s
+            securityContext: {}
+            serviceMonitorSelector:
+              matchLabels:
+                lab: custom-hpa
+
+    - name: Wait until service is available
+      k8s_facts:
+        kind: Service
+        name: prometheus-operated
+        namespace: my-prometheus
+      register: r_prom_svc
+      until:
+      - prom_svc.resources | length > 0
+
+    - name: Expose the service
+      command: "oc expose svc prometheus-operated -n my-prometheus"
+
+    - name: Create my-prometheus-hpa RoleBinding
+      k8s:
+        state: present
+        definition:
+          kind: RoleBinding
+          apiVersion: rbac.authorization.k8s.io/v1
+          metadata:
+            name: my-prometheus-hpa
+            namespace: my-hpa
+          subjects:
+            - kind: ServiceAccount
+              name: prometheus-k8s
+              namespace: my-prometheus
+          roleRef:
+            apiGroup: rbac.authorization.k8s.io
+            kind: ClusterRole
+            name: view
+
+    - name: Create custom k8s adapter
+      command: "oc create -f https://raw.githubusercontent.com/redhat-gpte-devopsautomation/ocp_advanced_deployment_resources/master/ocp4_adv_deploy_lab/custom_hpa/custom_adapter_kube_objects.yaml"
+
+    - name: Wait until custom adapter is available
+      k8s_facts:
+        api_version: apps/v1
+        kind: Deployment
+        name: custom-metrics-apiserver
+        namespace: my-prometheus
+      register: r_custom_adapter
+      until:
+      - r_custom_adapter | length > 0
```

Comment:
I think this is a weak check. You are looking to see if a Deployment exists, but not the resulting objects (i.e. Pod(s)) that the Deployment is creating or the functionality they provide. As a solver, each of these tasks should be as close as possible to the task that a user performs. In the current version of this lab, they would be running this command:
```
oc get --raw /apis/custom.metrics.k8s.io/v1beta1/ | jq -r '.resources[] | select(.name | contains("pods/http"))'
```
This should be what you are checking to make sure the result is as expected.

Reply:
Sounds reasonable let me figure out how to implement this, thanks for pointing

Reply:
Resolved on next commit.